src/RSocketClient.h

```diff
@@ -56,13 +60,26 @@ class RSocketClient {
       std::unique_ptr<DuplexConnection> connection,
       folly::EventBase& eventBase,
       SetupParameters setupParameters = SetupParameters(),
-      std::shared_ptr<RSocketResponder> responder = std::shared_ptr<RSocketResponder>(),
-      std::unique_ptr<KeepaliveTimer> keepaliveTimer = std::unique_ptr<KeepaliveTimer>(),
+      std::shared_ptr<RSocketResponder> responder =
+          std::shared_ptr<RSocketResponder>(),
+      std::unique_ptr<KeepaliveTimer> keepaliveTimer =
+          std::unique_ptr<KeepaliveTimer>(),
       std::shared_ptr<RSocketStats> stats = std::shared_ptr<RSocketStats>(),
-      std::shared_ptr<RSocketNetworkStats> networkStats = std::shared_ptr<RSocketNetworkStats>());
+      std::shared_ptr<RSocketNetworkStats> networkStats =
+          std::shared_ptr<RSocketNetworkStats>());
+
+  folly::Future<folly::Unit> resume(
+      std::unique_ptr<ClientResumeStatusCallback> resumeCallback =
+          std::make_unique<DefaultResumeCallback>());
+
+  void disconnect(folly::exception_wrapper ex);
 
  private:
   std::unique_ptr<ConnectionFactory> connectionFactory_;
   std::unique_ptr<RSocketConnectionManager> connectionManager_;
+
+  std::shared_ptr<RSocketStateMachine> stateMachine_;
```

Comment:
Prior to this change (and after this change) the application would need to store both RSocketClient and RSocketRequester.  The RSocketClient holds ConnectionManager instance.  The ConnectionManager destructor closes and destroys all RSocketStateMachines.  This design seems intentional.  So the application has to hold the RSocketClient instance in addition to the RSocketRequester.   

But I agree  that the RSocketClient-RSocketRequester interaction/ownership is kind of awkward here.  The clients shouldn't have to worry about life-times of both these objects.  Maybe RSocketClient can hold a instance of RSocketRequester?  That way, the client needs to worry about only holding the RSocketClient.